### ISSUE TITLE
Class "Magento\Framework\Mview\View\ChangeLogBatchWalker" does not exist

### ISSUE THREAD

**Issue by Quazz** [magento/meta-for-magento2#88]
<!---
Fields marked with (*) are required. Please don't remove the template.
-->

### Preconditions (*)
<!---
Provide the exact extension version and any important information on the environment where bug is reproducible.
-->
1. Magento 2.4.7-p1
2. Meta 1.3.3

### Steps to reproduce (*)
<!---
Important: Provide a set of clear steps to reproduce this bug. We can not provide support without clear instructions on how to reproduce.
-->
1. Install and configure the extension
2.  Wait for indexing

### Expected result (*)
<!--- Tell us what do you expect to happen. -->
1. Reindexes, no errors

### Actual result (*)
<!--- Tell us what happened instead. Include error messages and issues. -->
1.  Reindex fails because it can't find the class



There was an attempt to support both the older and current versions of Magento.

Unfortunately the implementation does not work currently.

https://github.com/magento/meta-for-magento2/blob/2b92fa9362da29f2ccd62a73bb0072a88c8e0e4f/app/code/Meta/Catalog/Plugin/FacebookCatalogUpdateFullReindexPlugin.php#L105

The issue is on this line.

The problem is that class_exists PHP function searches for the class in a case insensitive manner and will thus always return ChangeLogBatchWalker and never ChangelogBatchWalker. (in fact PHP treats classes in case insensitive manner in general, but the autoloader does not)

My suggestion would be something like this:

```
        $changeLogWalkerFactory = $this->objectManager->create(
            \Magento\Framework\Mview\View\ChangeLogBatchWalkerFactory::class // @phpstan-ignore-line
        );
        if (get_class($changeLogWalkerFactory) == "ChangeLogBatchWalkerFactory") {
            return $changeLogWalkerFactory->create(
                \Magento\Framework\Mview\View\ChangeLogBatchWalker::class // @phpstan-ignore-line
            );
        }

        $changelogWalkerFactory = $this->objectManager->create(
            \Magento\Framework\Mview\View\ChangelogBatchWalkerFactory::class // @phpstan-ignore-line
        );
        return $changelogWalkerFactory->create(
            \Magento\Framework\Mview\View\ChangelogBatchWalker::class // @phpstan-ignore-line
        );
```

get_class will return the actual class name in use, case sensitive.

**Comment by aarontalcott** [comment]
Seeing the same issue with 2.4.7-p3 and 1.3.4.

**Comment by sol-loup** [comment]
Confirmed:

Design Config Grid index has been rebuilt successfully in 00:00:00
Customer Grid index has been rebuilt successfully in 00:00:00
Category Products index has been rebuilt successfully in 00:00:03
Product Categories index has been rebuilt successfully in 00:00:00
Catalog Rule Product index has been rebuilt successfully in 00:00:00
Product EAV index has been rebuilt successfully in 00:00:00
Stock index has been rebuilt successfully in 00:00:00
Inventory index has been rebuilt successfully in 00:00:00
Catalog Product Rule index has been rebuilt successfully in 00:00:00
Product Price index has been rebuilt successfully in 00:00:00
Catalog Search index has been rebuilt successfully in 00:00:10
Stores Feed index has been rebuilt successfully in 00:00:00
Sales Order Feed index has been rebuilt successfully in 00:00:00
Sales Order Statuses Feed index has been rebuilt successfully in 00:00:00

Added into our next build. Thanks @Quazz !

**Comment by zlik** [comment]
Fixed in [1.3.5](https://github.com/magento/meta-for-magento2/releases/tag/v1.3.5), closing the issue.

**Comment by AlexandreBulete** [comment]
Hi, your fix is incomplete.
The php function get_class returns the full name of the class (with its namespace), so your condition should be : 
```php
if (get_class($changeLogWalkerFactory) == "Magento\Framework\Mview\\NChangeLogBatchWalkerFactory")
```
